utils/errors.js

```diff
@@ -1,9 +1,12 @@
 function createError(status, name) {
-  function NewError(code, message) {
+  function NewError(code, message, data) {
     this.code = code;
     this.message = message;
     this.status = status;
     this.name = name;
+    if (data) {
+      this.data = data;
+    }
```

Comment:
I didn't want to use simply `this.data = data` (without the condition) just by correctness.